esti/migrate_test.go

```diff
@@ -542,3 +554,147 @@ func verifyUserPermissions(t *testing.T, ctx context.Context, repo, userType str
+	})
+	require.NoError(t, err, "failed to create tag")
+	require.Equal(t, http.StatusCreated, createTagResp.StatusCode())
+
+	// create branch with uncommitted data for the post test
+	createBranchResp, err := client.CreateBranchWithResponse(ctx, repo, api.CreateBranchJSONRequestBody{
+		Name:   "pre_branch",
+		Source: mainBranch,
+	})
+	require.NoError(t, err, "failed to create branch")
+	require.Equal(t, http.StatusCreated, createBranchResp.StatusCode())
+
+	uploadFiles(t, ctx, repo, "pre_branch", "b/foo/")
+
+	// update state
+	state.Graveler.Repo = repo
+	state.Graveler.CommitID = commitResp.JSON201.Id
+	state.Graveler.TagID = "pre_tag"
+	state.Graveler.committedObjPath = "a/foo/1"
+	state.Graveler.uncommittedObjPath = "b/foo/1"
+}
+
+func testPostMigrateGraveler(t *testing.T) {
+	ctx, _, _ := setupTest(t)
+
+	// verifying all previous resources are preserved through the migration process
+	verifyGravelerEntities(t, ctx)
+
+	// commit to the uncommitted branch
+	commitResp, err := client.CommitWithResponse(ctx, state.Graveler.Repo, "pre_branch", &api.CommitParams{}, api.CommitJSONRequestBody{
+		Message: "commit after migrate",
+	})
+	require.NoError(t, err, "failed to commit changes")
+	require.Equal(t, http.StatusCreated, commitResp.StatusCode())
+
+	// create tag from previous commit
+	createTagResp, err := client.CreateTagWithResponse(ctx, state.Graveler.Repo, api.CreateTagJSONRequestBody{
+		Id:  "post_tag",
+		Ref: state.Graveler.CommitID,
+	})
+	require.NoError(t, err, "failed to create tag")
+	require.Equal(t, http.StatusCreated, createTagResp.StatusCode())
+
+	// create branch from the previous commit
+	createBranchResp, err := client.CreateBranchWithResponse(ctx, state.Graveler.Repo, api.CreateBranchJSONRequestBody{
+		Name:   "post_branch",
+		Source: state.Graveler.CommitID,
+	})
+	require.NoError(t, err, "failed to create branch")
+	require.Equal(t, http.StatusCreated, createBranchResp.StatusCode())
+
+	uploadFiles(t, ctx, state.Graveler.Repo, "post_branch", "c/foo/")
+
+	// revert
+	revertResp, err := client.RevertBranchWithResponse(ctx, state.Graveler.Repo, mainBranch, api.RevertBranchJSONRequestBody{
+		Ref: mainBranch,
+	})
+	require.NoError(t, err, "failed to revert")
+	require.Equal(t, http.StatusCreated, revertResp.StatusCode())
+
+	// assert file doesn't exist
+	unobjResp, err := client.GetObjectWithResponse(ctx, state.Graveler.Repo, mainBranch, &api.GetObjectParams{Path: state.Graveler.committedObjPath})
+	require.NoError(t, err)
+	require.Equal(t, unobjResp.HTTPResponse.StatusCode, http.StatusNotFound, "object not found")
+
+	verifyGravelerList(t, ctx)
+}
+
+func uploadFiles(t *testing.T, ctx context.Context, repo string, branch string, prefix string) {
+	for i := 0; i < migrateFileSize; i++ {
+		_, _, err := uploadFileRandomDataAndReport(ctx, repo, branch, prefix+strconv.Itoa(i), true)
+		if err != nil {
+			t.Error(err)
+		}
+	}
+}
+
+func verifyGravelerEntities(t *testing.T, ctx context.Context) {
```

Comment:
Please also validate the data of these fields and not just existence.

Reply:
I'm validating it on the graveler migrate unit test